src/plugins/crypto/gcrypt_operations.c

```diff
@@ -0,0 +1,243 @@
+/**
+* @file
+*
+* @brief cryptographic interface using the gcrypt library
+*
+* @copyright BSD License (see doc/COPYING or http://www.libelektra.org)
+*
+*/
+
+#include "crypto.h"
+#include "gcrypt_operations.h"
+#include <kdberrors.h>
+#include <stdlib.h>
+#include <gcrypt.h>
+
+
+static void addPkcs7Padding(unsigned char *buffer, const unsigned int contentLen, const unsigned int bufferLen);
+static unsigned int getPkcs7PaddedContentLen(const unsigned char *buffer, const unsigned int bufferLen);
+
+
+void elektraCryptoGcryHandleDestroy(elektraCryptoHandle *handle)
+{
+	if(handle != NULL)
+	{
+		gcry_cipher_close(*handle);
+		free(handle);
+	}
+}
+
+int elektraCryptoGcryInit()
+{
+	if (!gcry_check_version(GCRYPT_VERSION))
+	{
+		ELEKTRA_SET_ERRORF(111, NULL, "Libgcrypt version check failed, looking for version: %s", GCRYPT_VERSION);
```

Comment:
With NULL as argument nothing will happen. You need to supply a key where the warning/error is attached.